### ISSUE TITLE
Changed URLS?

### ISSUE THREAD

**Issue by oddobert** [isometric/EverWingHax#5]
Hi,

Just tried to run the script on my machine.
Thank you for a well documented approach, 

There is one issue, I think the urls have changed. 
Following your explanation in the script I find the only active appspot url to be: wintermute-151001.appspot.com
Furthermore no "/auth/" - link is found in the network log. It's now:
https://wintermute-151001.appspot.com/game/state/everwing/default/uid
https://wintermute-151001.appspot.com/game/action
https://wintermute-151001.appspot.com/game/time
https://wintermute-151001.appspot.com/game/state/everwing/bossRaid_uid
https://wintermute-151001.appspot.com/game/context/shared/everwing/
-Where uid is my user id.

Looking through your script I also can not identify a JSON response in the network log that matches the expected input. 

TL;DR: Is the script outdated due to changes to Everwing? 

Thank you for any response you may have.

**Comment by oddobert** [comment]
HI again,

I have made some progress.
The stats are now compressed in an lz-string (python version for decompressing found here: https://github.com/marcel-dancak/lz-string-python).
The JSON is found there. So the script needs an additional decompress / compress cycle to work. The urls also needs to be updated.

**Comment by Saturate** [comment]
It looks like the object has changed as well. You can test it out on: https://codepen.io/Saturate/pen/ybvvbP

Its' the entire state that gets posted.

**Comment by IPFR33LY** [comment]
I'll take a look at this tomorrow. See if there's anything else that had changed.

**Comment by IPFR33LY** [comment]
Alright,

I have a working version that will take all your current dragons and fully upgrade them. It can also MAX out all currencies, and player XP.

It can also unlock Jade and all 'LOCKED' characters.

All it requires is a players User ID, which can be found by going into devtools (chrome: F12) and watching for a XHR request. You can filter for /action, and it'll have your User ID.

If anyone has a User ID they'd like me to test it on, I'd be more then happy to.

Also, found several exploits that cause severe denial of service to any facebook. It would keep them from ever playing Everwing again.

Edit: It might be possible to unlock all possible dragons, but I am unsure about the 'value' and 'maximum' attributes under the 'sidekick:$$_zodiac' instance. Not sure if that means the number of dragons with that zodiac or not.

**Comment by alexratman** [comment]
Can you share the working version. I'd rather see the code than share a user ID. Thanks

**Comment by carlthome** [comment]
Interesting stuff. Just compressing the JSON to obfuscate game state seems insane to me. If they ever were to monetize the game, surely they would need to at least stop allowing POSTs from anything, better yet use HTTPS. @IPFR33LY feel free to run it with 1688035771210006 for testing. I have no moral qualms.

**Comment by IPFR33LY** [comment]
uID 1688035771210006 does not exist on their server. I didn't think of it but maybe I can have it default to uploading a "Generic state" if a current "game state" doesn't exist for a user.

**Comment by carlthome** [comment]
"1688035771210006" is definitely in the JSON request payload to wintermute-151001.appspot.com on my end. The same goes for `bossRaid_x/` URLs, where the user id is appended, e.g. `https://wintermute-151001.appspot.com/game/state/everwing/bossRaid_1313036592066372/1688035771210006`

Maybe the server also checks that the user id matches the logged in Facebook account somehow, or just responds that it doesn't exist otherwise. Have you tried with other ids aside from your own?

**Comment by IPFR33LY** [comment]
Yeah, it's indeed the right one. I was attempting to copy+paste from my phone last night to a teamviewer session, when I was out and about.

You should see your dragons updated, all characters unlocked and currencies set to max.

**Comment by IPFR33LY** [comment]
https://gist.githubusercontent.com/IPFR33LY/1284894c871b65b69f5c2ee452a554cd/raw/764bb2a7a5853c2e3c33c24f70a28e96639af071/EverwingHax-001
 
You'll need `https://github.com/eduardtomasek/lz-string-python`

Created this with Python 2.7, the lz-string-python was created for Python 3 but works with 2.7.

The script also outputs the entirety of your state into a text file called `statefile.txt` in the same directory of the script. This statefile will have all the updated changes done wrote to it. This is just for informational purposes.

**Comment by pcaande** [comment]
@IPFR33LY Getting error below

File "Path/EverwingHax-001.py", line 18, in <module>
    from p3lzstring import LZString
ImportError: No module named p3lzstring

Changed p3lzstring to lzstring and i get this error,

File "PATH/EverwingHax-001.py", line 164, in <module>
    E=O(Q(E))
  File "PATH/EverwingHax-001.py", line 32, in O
    J=LZString.decompressFromBase64(lzstring)
TypeError: unbound method decompressFromBase64() must be called with LZString instance as first argument (got str instance instead)
>>>

Can you tell me what im doing wrong?

**Comment by IPFR33LY** [comment]
Yeah. Once I'm off mobile, I'll double check it.

**Comment by jinyu1993** [comment]
@IPFR33LY I am getting this error when i input Y to level all dragon to max, money and character works fine
Traceback (most recent call last):
  File "C:\Users\User\Downloads\EverwingHax-001.py", line 169, in <module>
    v(R(j(w),w))
  File "C:\Users\User\Downloads\EverwingHax-001.py", line 114, in v
    c(i,o,i,maturity='Maturity',XP=2)
  File "C:\Users\User\Downloads\EverwingHax-001.py", line 99, in c
    w['instances'][H]=o[x][maturity]
TypeError: list indices must be integers, not unicode

Been trying to mess with the code with no avail

**Comment by IPFR33LY** [comment]
@jinyu1993 @oddobert @pcaande @carlthome @alexratman 
Here, I've uploaded it to a repository.

https://github.com/IPFR33LY/EverwingHax

If you have additional changes.. formatting.. Whatever, improvements... Submit pull requests there.

**Comment by ellemaker** [comment]
Traceback (most recent call last):
  File "EverWingHax.py", line 423, in <module>
    main()
  File "EverWingHax.py", line 76, in main
    acquire_characters()
  File "EverWingHax.py", line 94, in acquire_characters
    characters = get_item_class("character")
  File "EverWingHax.py", line 346, in get_item_class
    return[item for item in world["player"]["inventory"] if type_name in item["model"]]
KeyError: 'player'


got this error

**Comment by IPFR33LY** [comment]
@ellemaker This script doesn't work anymore. Check out my link above.
